### PR TITLE
PHP7.2 & strict type hinting

### PR DESCRIPTION
**Pull request by odolbeau**
Before releasing major version of the symfony storage, extractor & so on, let's start by the beginning: this library!

This PR:
* Remove support of PHP < 7.2
* Add strict I/O type hinting
* Add github actions (phpstan & php-cs-fixer)
* Remove StyleCI
* Prepare release 2.0.0

### CODE REVIEW

File: .github/workflows/static.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+on: [push, pull_request]
```
Comment (odolbeau) [inline .github/workflows/static.yml:1]: I used exactly the same config as the Symfony bundle.

File: phpstan.neon.dist [hunk 2]
```diff
@@ -0,0 +1,4 @@
+parameters:
+    level: 5
```
Comment (odolbeau) [inline phpstan.neon.dist:2]: level 6 is failing due to missing strict type hinting on some method, ie in `MessageInterface`:
```php
    /**
     * @param mixed|null $default
     *
     * @return mixed|null
     */
    public function getMeta(string $key, $default = null);
```

**Comment by odolbeau** [comment]
@Nyholm It looks like this repository doesn't have the same configuration as others (no mandatory review).
Could you please correct this? Thanks! :)

**Review by Nyholm — APPROVED** [review]
Thank you.

**Pull request merged**